yaspin/core.py

```diff
@@ -436,6 +447,16 @@ def _compose_color_func(self) -> Optional[Callable[..., str]]:
     def _compose_out(self, frame: str, mode: Optional[str] = None) -> str:
         text = str(self._text)
 
+        term_width = shutil.get_terminal_size()[0]
+
+        # Width - (Frame + Space) - (Timer + Space) - Ellipsis
+        max_text_len = term_width - (len(frame) + 1)
+        max_text_len = max_text_len - (0 if not self._timer else 12 + 1)
```

Comment:
Where is this `12 + 1` comes from? If this is a timer formatter from:
```
            text += " ({}.{:02.0f})".format(  # pylint: disable=consider-using-f-string
                timedelta(seconds=sec), fsec
            )
```
let's abstract it away into a dedicated variable and apply `len()` on it here.